lib/cursor/core_cursor.js

```diff
@@ -434,6 +434,10 @@ class CoreCursor extends Readable {
     }
 
     function done(err, result) {
+      if (err && cursor._endSession) {
```

Comment:
After I monkey patched mongodb with `processWaitQueue` and deployed in my environment, when this happen again, the queue is no more memory leak if I set `waitQueueTimeoutMS`. But there's another place where there's a memory leak.

Then I take a heapdump

![image](https://user-images.githubusercontent.com/985607/82476481-bc51bc80-9b00-11ea-8e6f-c76c1e8c2d95.png)
![image](https://user-images.githubusercontent.com/985607/82476588-e1462f80-9b00-11ea-9a4b-5b651835c80e.png)

But I'm not sure if we can fix this memory leak like this.